src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,586 @@
+                    z: -1
+                    opacity: 0.8
+                    gradient: Gradient {
+                        GradientStop { position: 0.0; color: "#000" }
+                        GradientStop { position: 0.9; color: "#666" }
+                        GradientStop { position: 1.0; color: "#AAA" }
+                    }
+                    radius: 10
+                    border.color: 'black'
+                    border.width: 1
+                    anchors.centerIn: parent
+                    width: parent.width * 1.1
+                    height: parent.height
+                }
+                property string value: first_screen.visible ? '' : items.toAlpha ?
+                                           morseConverter.morse.trim() :
+                                           morseConverter.alpha ? morseConverter.alpha : ''
+            }
+
+            Timer {
+                id: timer
+                interval: 500
+                onTriggered: {
+                    if( feedback.value == items.questionValue ) {
+                        bonus.good('tux')
+                        textInput.clear()
+                    }
+
+                }
+            }
+            Item {
+                id:margin
+                width: 1
+                height: 20 * ApplicationInfo.ratio
+            }
+
+            GCText {
+                id: instruction
+
+                fontSize: largeSize
+                minimumPixelSize: 70
+                fontSizeMode: Text.Fit
+                horizontalAlignment: Text.AlignHCenter
+                anchors.horizontalCenter: parent.horizontalCenter
+                width: parent.width * 0.5
+                text: items.instruction
+                height: Text.Fit
+                wrapMode: TextEdit.WordWrap
+                color: 'black'
+
+            }
+
+
+        }
+
+        Rectangle {
+            id: morse_map
+            width: parent.width * 0.8
+            height: parent.height * 0.7
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.verticalCenter: parent.verticalCenter
+            visible: false
+
+            Flow {
+                id: grid1
+                width: parent.width * 0.9
+                anchors {
+                    top: parent.top
+                    topMargin: 10 * ApplicationInfo.ratio
+                    horizontalCenter: parent.horizontalCenter
+                }
+                spacing: 5 * ApplicationInfo.ratio
+
+                Repeater {
+                    id: cardRepeater
+                    model: [
+                        "A", "B", "C", "D", "E", "F", "G", "H", "I", "J",
+                        "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T",
+                        "U", "V", "W", "X", "Y", "Z", "1", "2", "3", "4",
+                        "5", "6", "7", "8", "9", "0"
+                    ]
+
+                    Column {
+                        width:  (grid1.width - grid1.spacing * 10) / 10
```

Comment:
you are forcing to have 10 elements per Row doing like this. You should let Qt/Qml Flow computes by itself how many elements are needed per Row.
You need to specify the element's width and adjust the Column width with it, not the opposite. Then, if the height of the items is bigger than the morse_map Rectangle height, you can add a Flickable to be able to navigate in it.
Don't forget to take in account the bar height and the virtual keyboard height (when displayed) too.